docs(grid): Add Cell Selection documentation and revamp Selection articles (#2211)
telerik/blazor-docs: components/grid/selection/overview.md

# Grid Selection

> The Grid component offers support for row and cell selection. 

Reviewer: ```suggestion
The Grid component supports row and cell selection.
```

> The Grid component offers support for row and cell selection. 
> 
> In this article:

Reviewer: ```suggestion
The Grid component supports row and cell selection. This article provides an overview of the Grid selection behavior and configuration:
```

* [Selection Type](#selection-type)
* [Selection Mode](#selection-mode)
* [Selection Options](#selection-options)
* [Selected Items](#selected-items)
* [Selection Events](#selection-events)

> * [Selection and Other Grid Features](#selection-and-other-grid-features)

Reviewer: These link labels are too brief and the uneducated reader will not benefit from them. Consider more detailed descriptions like in http://127.0.0.1:4000/blazor-ui/common-features/data-binding/onread

There is no requirement for these links to match the headings below. We already have that in the right navigation, so it brings no added value to repeat that.

## Selection Type

You can configure the Grid either for row or cell selection:

> * To allow row selection:

Reviewer: ```suggestion
* To enable row selection:
```

  * Set the [Grid `SelectionMode` parameter](#selection-mode) or

>   * Add a `<GridSelectionSettings>` tag to the `<GridSettings>` tag, and set the `SelectionType` parameter to the `Row` member of the `Telerik.Blazor.GridSelectionType` enum

Reviewer: ```suggestion
  * Add a `<GridSelectionSettings>` tag to the `<GridSettings>` tag, and set the `SelectionType` parameter to `Telerik.Blazor.GridSelectionType.Row`
```

* To allow cell selection:
  * Add a `<GridSelectionSettings>` tag to the `<GridSettings>` tag, and set the `SelectionType` parameter to the `Cell` member of the `Telerik.Blazor.GridSelectionType` enum

## Selection Mode

You can configure the selection behavior by setting the Grid `SelectionMode` parameter to a member of the `Telerik.Blazor.GridSelectionMode` enum. The Grid supports the following selection modes:

> * `None` (the default value)—To disable row or cell selection.

Reviewer: ```suggestion
* `None` (default)—Disables row and cell selection.
```

* `Single`—Allows the user to select only one cell or row at a time. If the user attempts to select multiple cells or rows sequentially, only the most recent selection will be retained.
* `Multiple`—Allows the user to select multiple rows or cells at a time.

## Selection Options

> When you select a row or a cell, they will be highlighted to notify you that they are selected. By default you can click on a row or on a cell to select it. You can also select a row through a checkbox column. To select multiple rows or cells, use the `Ctrl` or `Shift` key to extend the selection.

Reviewer: * Is " to notify you that they are selected" necessary?
* Link the checkbox column article.

See [Rows Selection Options]({%slug components/grid/selection/rows%}#rows-selection-options) and [Cells Selection Options]({%slug components/grid/selection/cells%}#cells-selection-options) for more details.

> ## Selected Items

Reviewer: This heading concerns only half of the section content. I agree with Dancho about the briefness of the sections as a whole.

You can get and set the selected rows or cells:
* To access the selected rows add the Grid `SelectedItems` parameter.
* To access the selected cells add the Grid `SelectedCells` parameter.

Both parameters are collections. You can use the parameters to manipulate the selected rows or cells. You can also use the parameters to pre-select a row or cell for your users.

See [Selected Rows]({%slug components/grid/selection/rows%}#selected-rows) and [Selected Cells]({%slug components/grid/selection/cells%}#selected-cells) for more details.

## Selection Events

You can respond to the user action of selecting a new item through the Grid events:
* Use the [`SelectedItemsChanged` event]({%slug components/grid/selection/rows%}#selecteditemschanged) to respond to row selection.
* Use the [`SelectedCellsChanged` event]({%slug components/grid/selection/cells%}#selectedcellschanged) to respond to cell selection.

> > Both binding to the property and using its event cannot be used at the same time as Blazor allows only one. To use the `SelectedItemsChanged` event, utilize one-way binding for the `SelectedItems` property. Otherwise, use two-way binding for the `SelectedItems` property without the `SelectedItemsChanged` event. The same applies to the `SelectedCells` and the `SelectedCellsChanged`.

Reviewer: This is simple general knowledge and we should include it only if we are sure that customers will stumble upon this.

If you decide to leave the note, it needs simplification and refactoring.

## Selection and Other Grid Features

The selection feature behavior may vary when the Grid configuration combines row or cell selection and:
* The [Grid editing feature]({%slug components/grid/editing/overview%}).
* The [Grid virtualization feature]({%slug components/grid/virtual-scrolling%}).
* The [Grid paging feature]({%slug components/grid/features/paging%}).
* The [Grid templates feature]({%slug components/grid/features/templates%}).

> * The [Grid drag and drop feature]({%slug grid-drag-drop-overview%}).

Reviewer: These links imply that the linked pages contain information about integration with the selection feature, which they don't. You can remove the whole list and just leave the two links at the end.

There are some limitations or additional modifications that are needed in such cases.

> See [Rows Selection and Other Grid Features]({%slug components/grid/selection/rows%}#rows-selection-and-other-grid-features) and [Cells Selection and Other Grid Features]({%slug components/grid/selection/cells%}#cells-selection-and-other-grid-features) for more details.

Reviewer: ```suggestion
See [Rows Selection and Other Grid Features]({%slug components/grid/selection/rows%}#row-selection-and-other-grid-features) and [Cells Selection and Other Grid Features]({%slug components/grid/selection/cells%}#cell-selection-and-other-grid-features) for more details.
```
